astropy/wcs/wcs.py

```diff
@@ -438,13 +439,14 @@ def __init__(
             sip = None
         else:
             keysel_flags = _parse_keysel(keysel)
-
-            if isinstance(header, (str, bytes)):
+            if isinstance(header, (str, bytes, os.PathLike)):
                 try:
-                    is_path = os.path.exists(header)
+                    is_path = os.path.exists(header)  # noqa: PTH110
                 except (OSError, ValueError):
                     is_path = False
-
+                # FIXME: this shouldn't be needed
+                if isinstance(header, os.PathLike):
+                    header = str(header)
```

Comment:
why is this here if it is not needed? I guess, I do not understand the comment.

Reply:
Sorry I should have been clear in the comment, in an ideal world, we should never really need to do cast Path objects to strings but a lot of code in astropy was written to work with strings.

It's becoming a big PR and I was worried about changing too many things at once. I plan on either fixing it in this PR or leave a much more informative FIXME for the next one :)